python/paddle/fluid/framework.py

```diff
@@ -4752,9 +4752,11 @@ def _dygraph_guard(tracer):
     global _dygraph_tracer_
     tmp_trace = _dygraph_tracer_
     _dygraph_tracer_ = tracer
+    core._switch_tracer(tracer)
 
     yield
 
+    core._switch_tracer(None)
```

Comment:
Why switch to `None` instead of `tmp_trace`?

Reply:
a mistake, updated to `tmp_trace`